diff --git a/examples/mobile/src/lib.rs b/examples/mobile/src/lib.rs
index ee49c01c5..e71a027fc 100644
--- a/examples/mobile/src/lib.rs
+++ b/examples/mobile/src/lib.rs
@@ -175,8 +175,11 @@ fn setup_music(asset_server: Res<AssetServer>, mut commands: Commands) {
 // This is handled by the OS on iOS, but not on Android.
 fn handle_lifetime(
     mut lifecycle_events: EventReader<AppLifecycle>,
-    music_controller: Single<&AudioSink>,
+    music_controller: Option<Single<&AudioSink>>,
 ) {
+    let Some(music_controller) = music_controller else {
+        return;
+    };
     for event in lifecycle_events.read() {
         match event {
             AppLifecycle::Idle | AppLifecycle::WillSuspend | AppLifecycle::WillResume => {}
