diff --git a/.github/start-mobile-example/mobile.conf.js b/.github/start-mobile-example/mobile.conf.js
index ed0cf8d90..3443c3f71 100644
--- a/.github/start-mobile-example/mobile.conf.js
+++ b/.github/start-mobile-example/mobile.conf.js
@@ -15,8 +15,7 @@ exports.config = {
         device: process.env.DEVICE || 'Samsung Galaxy S23',
         os_version: process.env.OS_VERSION || "13.0",
         app: process.env.BROWSERSTACK_APP_ID,
-        'browserstack.debug': true,
-        orientation: 'LANDSCAPE'
+        'browserstack.debug': true
     }],
 
     logLevel: 'info',
