diff --git a/Cargo.toml b/Cargo.toml
index 7bd99fc3a..f6a64d097 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -3526,17 +3526,6 @@ description = "Illustrates making the window transparent and hiding the window d
 category = "Window"
 wasm = false

-[[example]]
-name = "window_settings"
-path = "examples/window/window_settings.rs"
-doc-scrape-examples = true
-
-[package.metadata.example.window_settings]
-name = "Window Settings"
-description = "Demonstrates customizing default window settings"
-category = "Window"
-wasm = true
-
 [[example]]
 name = "window_drag_move"
 path = "examples/window/window_drag_move.rs"
